portal/patient_self_checkin/step4.php

```diff
@@ -0,0 +1,102 @@
+<?php
+$ignoreAuth = true;
+require_once "../../interface/globals.php";
+require_once $GLOBALS['srcdir'].'/sql.inc';
+
+?>
+
+<!-- Latest compiled and minified CSS -->
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+
+<!-- Optional theme -->
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">
+
+<!-- Latest compiled and minified JavaScript -->
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
+
+<style>
+
+.btn-lg {
+
+  width: 150px;
+  height: 100px;
+  margin: 5;
+  font-size: 30px;
+}
+
+.col-sm-3 {
+
+    text-align: center;
+}
+
+h2 { 
+
+text-align: center; }
+
+ body {
+    font-family: sans-serif;
+    background-color: #638fd0;
+
+    background: -webkit-radial-gradient(circle, white, #638fd0);
+    background: -moz-radial-gradient(circle, white, #638fd0);
+  }
+
+</style>
+
+<?php
+if ($GLOBALS['self_checkin_enable']==1) {
+    // Store variable from previous step
+    $sex = htmlspecialchars($_GET['sex']);
+    $month = htmlspecialchars($_GET['month']);
+    $date = htmlspecialchars($_GET['date']);
+?>
+
+<h2>Select the FIRST letter of your SURNAME</h2>
+
+<div class="row">
+  <div class="col-sm-2"><a href="step5.php?surname=A&date=<?php echo $date; ?>&sex=<?php echo $sex; ?>&month=<?php echo $month; ?>" class="btn btn-lg btn-default">A</a></div>
+  <div class="col-sm-2"><a href="step5.php?surname=B&date=<?php echo $date; ?>&sex=<?php echo $sex; ?>&month=<?php echo $month; ?>" class="btn btn-lg btn-default">B</a></div>
```

Comment:
This is going to be a bit hard to translate the letter selection :)
(for example, if using chinese names; no biggie for now since I can't really think of a solution)